docs/automated-testing/unit-testing/README.md

```diff
@@ -179,6 +179,16 @@ Some common pitfalls when writing a unit test that are important to avoid:
   the system is not entirely testable. It is best to wrap third party API calls in interfaces or other structures so
   that they do not get invoked in unit tests.
 
+### Integration tests instead of unit tests
+
+In some situations it is worth considering to write integration tests instead of unit tests to provide full test coverage. The prerequisite for this approach to be successful is to have integration being able to execute at a speed comparable to that of unit tests both locally and in a CI environment.
+Most modern web frameworks like ASP.NET or Spring Boot combined with the right mocking or stubbing approach for external dependencies offer excellent capabilities to enable such scenarios for testing.
+
+Usually, integration tests prove that independently developed modules connect together as designed.
+In order to replace unit tests with integration tests, the test coverage of integration tests need to be extended to verify the correct behavior of the system as well. Instead of several unit tests needed to test a specific case of functionality of the system, one integration scenario is created that covers the entire flow. For example in case of an API, the received HTTP responses and their content are verified for each request in test. This covers both the integration between components of the API and the correctness of its business logic.
+
+Using efficient integration tests as a replacement for unit tests has the advantage of testing the system as a black box without any knowledge of its internals. Code refactoring has no impact on tests. Common testing techniques as TDD can be applied at a higher level which results in a development process that is driven by acceptance tests. Complex business logic and corner cases can be still tested with unit tests if necessary to prevent over-bloating the integration test suite.
```

Comment:
I like the black box mentality, changing the environment to test and see how the service would behave.